sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -899,21 +899,15 @@ object SparkSession {
 
         // No active nor global default session. Create a new one.
         val sparkContext = userSuppliedContext.getOrElse {
+          val sparkConf = new SparkConf()
+          SparkContext.getOrCreate(sparkConf)
```

Comment:
in order to remove https://github.com/apache/spark/pull/18172/files#diff-d91c284798f1c98bf03a31855e26d71cL938 , here I change the behavior to also setting `options` to `sc.conf` even if the `sparkContext` is supplied by users. This change is safe as `SparkSession.Builder.sparkContext` is private, and I checked all the callers to confirm that it's ok to do so.